### PR TITLE
Added options / feature flags for enabling and running GC sweep.

### PR DESCRIPTION
**Pull request by agarwal-navin**
[AB#176](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/176).

Added the following options and feature flags to enable and run GC sweep phase.
- sweepAllowed - Controls whether GC sweep phase is allowed on a document. Persisted in summary and does not change throughout the lifetime of the document. Attempt to set this without "gcAllowed" that enables overall GC will result in a usage error.
- Fluid.GarbageCollection.RunSweep - Feature flag that can be used to override whether sweep runs or not. If this is set, it takes precedence and sweepAllowed value is ignored. The idea behind this is to test sweep on real documents without enabling it for the document. While we build confidence in the sweep process, if this is not set, we won't run sweep even if "sweepAllowed" is true.

This adds a `sweepEnabled` property to metadata blob in summary and so, the snapshots in the test repo needs to be updated - https://github.com/microsoft/FluidFrameworkTestData/pull/44.

Following are the conditions under which sweep will run:
1. Overall GC (mark phase) must be enabled.
2. Session expiry and sweep should be enabled for this container. Without session expiry we cannot safely delete unreferenced objects. This condition (# 2) can be overridden via Fluid.GarbageCollection.RunSweep feature flag.

This PR also moved all the GC properties written into summary metadata into its own interface. GarbageCollector returns all of it together.

### CODE REVIEW

File: packages/runtime/container-runtime/src/garbageCollection.ts
Comment (tyler-cai-microsoft) [inline packages/runtime/container-runtime/src/garbageCollection.ts:448]: Why are you choosing the feature flag over the runtime option?

File: packages/runtime/container-runtime/src/garbageCollection.ts
Comment (tyler-cai-microsoft) [inline packages/runtime/container-runtime/src/garbageCollection.ts:293]: I'd like to not name this `sweepEnabled` it feels like if this is true, sweep should be running, maybe we should just be storing `sweepAllowed` based on the runtime option.

File: packages/runtime/container-runtime/src/summaryFormat.ts
Comment (tyler-cai-microsoft) [inline packages/runtime/container-runtime/src/summaryFormat.ts:110]: Same as above, maybe `sweepAllowed`, so that it sounds like `shouldRunGC` is the final say, and the RunSweep flag overrides it

File: packages/runtime/container-runtime/src/garbageCollection.ts
Comment (agarwal-navin) [reply packages/runtime/container-runtime/src/garbageCollection.ts:448]: It's explained in the PR description:
The idea behind this is to test sweep on real documents without enabling it for the document. While we build confidence in the sweep process, if this is not set, we won't run sweep even if "sweepAllowed" is true.

File: packages/runtime/container-runtime/src/garbageCollection.ts
Comment (agarwal-navin) [reply packages/runtime/container-runtime/src/garbageCollection.ts:293]: `sweepEnabled` does mean if its true, sweep will be running. Currently, we have another layer (feature flags) to enable sweep because we want to stage it such that we can test it, run sweep for a subset of audiences, etc. Basically, build confidence that sweep works as expected. Once we remove the feature flag, this is the source of truth for whether sweep runs or not.

If you mean just renaming the class member to `sweepAllowed` for now I can do that. But the property that we write in the metadata blob should be called `sweepEnabled` because we can't change that later.

**Review by tyler-cai-microsoft — APPROVED** [review]
LGTM, once https://github.com/microsoft/FluidFrameworkTestData/pull/44 is merged, update :)

**Pull request merged**
